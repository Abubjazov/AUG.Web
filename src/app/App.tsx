import { FC, lazy, Suspense } from 'react'

import { getAuth } from 'firebase/auth'
import WellcomePage from 'pages/WellcomePage'
import { Route, Routes } from 'react-router-dom'
import { useAppSelector } from 'store/hooks'
import { useAppDispatch } from 'store/hooks'
import { setAuthData, setUserAuthenticated } from 'store/slices/authSlice'
import Fallback from 'uikit/Fallback'

const App: FC = () => {
  const { isUserAuthenticated } = useAppSelector(state => state.auth)

  const dispatch = useAppDispatch()

  const auth = getAuth()

  auth.onAuthStateChanged(user => {
    if (user) {
      dispatch(
        setAuthData({
          uid: user.uid,
          email: user.email,
        }),
      )
      dispatch(setUserAuthenticated(true))
    } else {
      dispatch(setUserAuthenticated(false))

      dispatch(
        setAuthData({
          uid: undefined,
          email: null,
        }),
      )
    }
  })

  const Layout = lazy(() => import('pages/Layout'))

  const MainPage = lazy(() => import('pages/MainPage'))

  const SocialPage = lazy(() => import('pages/SocialPage'))

  const Page404 = lazy(() => import('pages/Page404'))

  return (
    <Suspense fallback={<Fallback />}>
      <Routes>
        <Route
          path="/"
          element={
            isUserAuthenticated ? (
              <Layout>
                <MainPage />
              </Layout>
            ) : (
              <WellcomePage />
            )
          }
        />

        <Route path="/social" element={<SocialPage />} />

        <Route path="*" element={<Page404 />} />
      </Routes>
    </Suspense>
  )
}

export default App
