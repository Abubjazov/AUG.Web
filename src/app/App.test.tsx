import { render } from '@testing-library/react'

import App from './App'

describe('App', () => {
  test('should render App default', () => {
    expect(render(<App />)).toMatchSnapshot()
  })
})
