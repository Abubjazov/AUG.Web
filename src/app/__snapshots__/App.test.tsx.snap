// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render App default 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="_root_0c8bf9"
        >
          Layout
          <div
            class="_root_9d7740"
          >
            <button
              class="_root_91d6ec _outlined_91d6ec "
              data-testid="base-button"
              type="button"
            >
              Open second content
            </button>
            <button
              class="_root_cc4fb1 _mobile_cc4fb1 _install_cc4fb1 "
              data-testid="install-button"
              type="button"
            >
              <svg
                fill="none"
                height="14"
                viewBox="0 0 14 14"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#clip0_36_2615)"
                >
                  <path
                    d="M13.3002 9.09998V11.9C13.3002 12.2713 13.1527 12.6274 12.8901 12.8899C12.6276 13.1525 12.2715 13.3 11.9002 13.3H2.1002C1.72889 13.3 1.3728 13.1525 1.11025 12.8899C0.847695 12.6274 0.700195 12.2713 0.700195 11.9V9.09998"
                    stroke="white"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M3.5 5.59998L7 9.09998L10.5 5.59998"
                    stroke="white"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M7 9.09995V0.699951"
                    stroke="white"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </g>
                <defs>
                  <clippath
                    id="clip0_36_2615"
                  >
                    <rect
                      fill="white"
                      height="14"
                      width="14"
                    />
                  </clippath>
                </defs>
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="_root_0c8bf9"
      >
        Layout
        <div
          class="_root_9d7740"
        >
          <button
            class="_root_91d6ec _outlined_91d6ec "
            data-testid="base-button"
            type="button"
          >
            Open second content
          </button>
          <button
            class="_root_cc4fb1 _mobile_cc4fb1 _install_cc4fb1 "
            data-testid="install-button"
            type="button"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_36_2615)"
              >
                <path
                  d="M13.3002 9.09998V11.9C13.3002 12.2713 13.1527 12.6274 12.8901 12.8899C12.6276 13.1525 12.2715 13.3 11.9002 13.3H2.1002C1.72889 13.3 1.3728 13.1525 1.11025 12.8899C0.847695 12.6274 0.700195 12.2713 0.700195 11.9V9.09998"
                  stroke="white"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  d="M3.5 5.59998L7 9.09998L10.5 5.59998"
                  stroke="white"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  d="M7 9.09995V0.699951"
                  stroke="white"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_36_2615"
                >
                  <rect
                    fill="white"
                    height="14"
                    width="14"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
